### PR TITLE
[PS5][Driver] Allow `-T` to override `--default-script`

### PR DESCRIPTION
**Pull request by playstation-edd**
If a linker script is explicitly supplied, there's no benefit to supplying a default script.

SIE tracker: TOOLCHAIN-17524

### CODE REVIEW

File: clang/test/Driver/ps5-linker.c
Comment (bd1976bris) [inline clang/test/Driver/ps5-linker.c:71]: NIT: -T/--script

**Review by bd1976bris — APPROVED** [review]
LGTM. I would have preferred a simpler way of checking that there are no other occurrences of -T/--script/--default-script but couldn't think of anything.

I didn't notice that this was an upstream review initially. It would be worth waiting for approval from Jeremy.

File: clang/test/Driver/ps5-linker.c
Comment (playstation-edd) [reply clang/test/Driver/ps5-linker.c:71]: There's no `--script` on `clang`. I may not understand your comment.

**Review by jmorse — APPROVED** [review]
LGTM

File: clang/test/Driver/ps5-linker.c
Comment (bd1976bris) [reply clang/test/Driver/ps5-linker.c:71]: No you haven't. I just assumed that there was, appologies.

**Pull request merged**
